src/styles/key-frame-styles.js

```diff
@@ -0,0 +1,25 @@
+import { css } from 'lit';
+
+export default css`
+  @keyframes spin {
+    0% { transform: rotate(0deg); }
+    100% { transform: rotate(360deg); }
+  }
+
+  /* Expand a block from 0 to something visible, then to probably full screen, then beyond most screens.
+     Unlike with a transition, the final CSS can have an unbounded height, which will take effect after the animation. */
+  @keyframes expand-height {
+    0% { max-height: 0; }
+    50% { max-height: 100px; }
+    95% { max-height: 1000px; }
+    100% { max-height: 5000px; }
+  }
+  /* Inverse of the above, collapsing quickly at first (to avoid a delay if the element is already quite short)
+     then slowing towards 0. */
+  @keyframes collapse-height {
```

Comment:
I think there is something off with the collapse, it waits until the animation of the rotating icon is done before collapsing, which makes the animation feel stuck. Any idea on what the difference is here?

![waits-until-animation-is-done](https://user-images.githubusercontent.com/5056218/225091862-207cea1d-f76d-4d1f-8ac9-1f7a90b557a1.gif)

Reply:
It's not technically waiting for anything, it's just that all the values of `max-height` from 5000px down to 100px or so don't make any visible difference, so you have to wait for all those animation frames before the ones that actually collapse it.

That's presumably why the existing `transition` has a negative delay (the `-1.0s`) which skips over that part of the transition. The `keyframes` tries to do something similar with the extra stops at 5% and 50%, so we can tweak those to be more aggressive if it still feels laggy - maybe `100px` should be more like 25%? The only tradeoff is making sure larger elements still look smooth.

Reply:
What do you think of this?

```css
@keyframes collapse-height {
    0% { max-height: 5000px; }
    5% { max-height: 500px; }
    40% { max-height: 100px; }
    85% { max-height: 0; }
  }
```

Reply:
Setting to 0 before reaching 100% feels weird - you can just make the absolute duration on the `animation` property shorter in that case.

Maybe we just need an extra key frame between 100px and 0? I've just shut my laptop down for the night so can't test, but halfway between your last two frames would be 50px at 62.5%, so maybe something like this?

```
@keyframes collapse-height {
    0% { max-height: 5000px; }
    5% { max-height: 500px; }
    40% { max-height: 100px; }
    60% { max-height: 50px; }
    100% { max-height: 0; }
}
```

Reply:
I thought of two ways to improve the feel of this: https://github.com/Rhosys/openapi-explorer/pull/143/commits/e5dbb7515124e92cb0340ac3b0b3b8221c399559

Have a play and see if that feels a bit more reactive.

Reply:
I think this is great, let's go with it.